src/topology/algebra/uniform_ring.lean

```diff
@@ -151,8 +152,32 @@ instance : comm_ring (completion R) :=
       (assume a b, by rw [← coe_mul, ← coe_mul, mul_comm]),
  ..completion.ring }
 
-end uniform_space.completion
+section algebra
+variables (S : Type*) [comm_semiring S] [algebra S R] [has_uniform_continuous_const_smul S R]
+
+@[simp] lemma map_smul_eq_mul_coe (s : S) :
+  completion.map ((•) s) = (*) (algebra_map S R s : completion R) :=
+begin
+  ext x,
+  refine completion.induction_on x _ (λ r, _),
+  { exact is_closed_eq (completion.continuous_map) (continuous_mul_left _) },
+  { rw [map_coe (has_uniform_continuous_const_smul.uniform_continuous_const_smul s),
+      algebra.smul_def, coe_mul];
+    apply_instance, },
+end
 
+instance : algebra S (completion R) :=
+{ commutes' := λ s x, by simp only [ring_hom.to_fun_eq_coe, mul_comm],
+  smul_def' := λ s x, congr_fun (map_smul_eq_mul_coe R S s) x,
+  ..((uniform_space.completion.coe_ring_hom : R →+* completion R).comp (algebra_map S R)) }
+
+lemma algebra_map_def (s : S) :
+  algebra_map S (completion R) s = (algebra_map S R s : completion R) :=
+rfl
+
```

Comment:
It's probably worth adding
```
/-- A shortcut instance for the common case -/
instance algebra' : algebra R (completion R) :=
by apply_instance
```
mainly because this confirms that the instance above really does generalize how we expect it to.